src/app/features/notification/notification-rule/notification-rule.component.ts

```diff
@@ -190,17 +191,57 @@ export class NotificationRuleComponent implements OnChanges {
     (this.form.get("conditions") as FormArray).push(newCondition);
   }
 
-  updateNotificationCondition(updateNotificationCondition: NotificationRule) {
-    this.notificationConditionValueChange.emit(updateNotificationCondition);
-  }
-
   removeCondition(notificationConditionIndex: number) {
     (this.form.get("conditions") as FormArray).removeAt(
       notificationConditionIndex,
     );
-    this.removeNotificationCondition.emit();
+    this.updateValue(this.form.value);
   }
 
+  /**
+   * Open the conditions JSON editor popup.
+   */
+  openConditionsInJsonEditorPopup() {
+    const notificationConditions = this.form.get("conditions")?.value;
+    const dialogRef = this.dialog.open(JsonEditorDialogComponent, {
+      data: {
+        value: this.parseConditionsArrayToObject(notificationConditions) ?? {},
+        closeButton: true,
+      },
+    });
+
+    dialogRef.afterClosed().subscribe((result) => {
+      this.handleConditionsJsonEditorPopupClose(result);
+    });
+  }
+
+  /**
+   * Handle the result of the conditions JSON editor popup.
+   * @param result
+   * @private
+   */
+  private handleConditionsJsonEditorPopupClose(result: string[]) {
+    if (!result) {
+      return;
+    }
+
+    const parsedObject = this.parseConditionsObjectToArray(result);
+    const conditionsFormArray = this.form.get("conditions") as FormArray;
+    conditionsFormArray.clear();
+    parsedObject.forEach((condition) => {
+      const conditionGroup = new FormGroup({
+        entityTypeField: new FormControl(condition.entityTypeField),
+        operator: new FormControl(condition.operator),
+        condition: new FormControl(condition.condition),
+      });
+      conditionsFormArray.push(conditionGroup);
+    });
```

Comment:
shouldn't this be more or less the same as the initForm() method that we call onChanges?

Reply:
I have updated the code and reused the init form control. Please take a look on the updated changes and let me know if there are any further changes needed.

Reply:
Thanks!